inject/inject.vala

```diff
@@ -306,6 +280,34 @@ namespace Frida.Inject {
 		}
 #endif
 
+		private void watch_stdin () {
+			/**
+			 * Support reading from stdin for communications with the injected script.
+			 * With the console in its default canonical mode, we will read a line at a
+			 * time when the user presses enter and send it to a registered RPC method
+			 * in the script as follows. Here, the data parameter is the string typed
+			 * by the user including the newline.
+			 *
+			 * rpc.exports = {
+			 *   onFridaStdin(data) {
+			 *     ...
+			 *   }
+			 * };
+			 */
+			var fd = stdin.fileno ();
+#if WINDOWS
+			var inchan = new IOChannel.win32_new_fd (fd);
+#else
+			var inchan = new IOChannel.unix_new (fd);
+#endif
+			inchan.add_watch (IOCondition.IN, (source, condition) => {
+				if (script_runner.terminal_mode == COOKED)
+					return read_line (source, condition);
+				else
+					return read_raw (source, condition);
+			});
+		}
```

Comment:
This is lower level than `start ()`, so we should move it down – I'd say all the way down inside this class. The methods it relies on should also be moved down, below it. (Following [Newspaper Code Structure](https://kentcdodds.com/blog/newspaper-code-structure/)).